airbyte-integrations/connectors/destination-clickhouse/src/main/java/io/airbyte/integrations/destination/clickhouse/ClickhouseDestination.java

```diff
@@ -81,26 +81,13 @@ public JsonNode toJdbcConfig(final JsonNode config) {
   }
 
   @Override
-  public AirbyteConnectionStatus check(final JsonNode config) {
-    final DataSource dataSource = getDataSource(config);
-    try {
-      final JdbcDatabase database = getDatabase(dataSource);
-      final NamingConventionTransformer namingResolver = getNamingResolver();
-      final String outputSchema = namingResolver.getIdentifier(config.get(JdbcUtils.DATABASE_KEY).asText());
-      attemptSQLCreateAndDropTableOperations(outputSchema, database, namingResolver, getSqlOperations());
-      return new AirbyteConnectionStatus().withStatus(Status.SUCCEEDED);
-    } catch (final Exception e) {
-      LOGGER.error("Exception while checking connection: ", e);
-      return new AirbyteConnectionStatus()
-          .withStatus(Status.FAILED)
-          .withMessage("Could not connect with provided configuration. \n" + e.getMessage());
-    } finally {
-      try {
-        DataSourceFactory.close(dataSource);
-      } catch (final Exception e) {
-        LOGGER.warn("Unable to close data source.", e);
-      }
+  protected AirbyteConnectionStatus checkedConnectionStatus(final DataSource dataSource, final JsonNode config, ConfiguredAirbyteCatalog catalog) throws Exception {
+    final JdbcDatabase database = getDatabase(dataSource);
+    final NamingConventionTransformer namingResolver = getNamingResolver();
+    for (final String outputSchema : getOutputSchemas(config, catalog)) {
+      attemptTableOperations(outputSchema, database, namingResolver, getSqlOperations(), false);
     }
+    return new AirbyteConnectionStatus().withStatus(Status.SUCCEEDED);
```

Comment:
I also noticed that for all these check methods here and below, we lost the datasource cleanup finally blocks. It's always nice to close the resources properly